sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        if (tokenCredential != null) {
+            // User token based policy
+            policies.add(new BearerTokenAuthenticationPolicy(tokenCredential, DEFAULT_SCOPE));
+        } else if (subscriptionKey != null) {
+            headers.put(OCP_APIM_SUBSCRIPTION_KEY, subscriptionKey);
+        } else {
+            // Throw exception that credential and tokenCredential cannot be null
+            logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
+        }
+
+        policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+            buildConfiguration));
+        policies.add(new RequestIdPolicy());
+        policies.add(new AddHeadersPolicy(headers));
+        policies.add(new AddDatePolicy());
+
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+        policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+
+        policies.addAll(this.policies);
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
+        policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+        // customized pipeline
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+
+        TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new TextAnalyticsAsyncClient(textAnalyticsAPI, serviceVersion);
+    }
+
+    /**
+     * Set the default client option for one client.
+     *
+     * @param clientOptions TextAnalyticsClientOptions model that includes
+     * {@link TextAnalyticsClientOptions#getDefaultLanguage() default language} and
+     * {@link TextAnalyticsClientOptions#getDefaultCountryHint() default country hint}
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder clientOptions(TextAnalyticsClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Text Analytics instance.
+     *
+     * @param endpoint The URL of the Azure Text Analytics instance service requests to and receive responses from.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
```

Comment:
You should have the ex as an inner throwable to that illegal argument exception, so consumers can understand why that parse failed.

Reply:
what is inner throwable?